app/mm-redux/actions/posts.ts

```diff
@@ -406,36 +418,50 @@ export function editPost(post: Post) {
 export function getUnreadPostData(unreadChan: ChannelUnread, state: GlobalState) {
     const member = getMyChannelMemberSelector(state, unreadChan.channel_id);
     const delta = member ? member.msg_count - unreadChan.msg_count : unreadChan.msg_count;
+    const deltaRoot = member ? member.msg_count_root - unreadChan.msg_count_root : unreadChan.msg_count_root;
 
     const data = {
         teamId: unreadChan.team_id,
         channelId: unreadChan.channel_id,
         msgCount: unreadChan.msg_count,
         mentionCount: unreadChan.mention_count,
+        msgCountRoot: unreadChan.msg_count_root,
+        mentionCountRoot: unreadChan.mention_count_root,
         lastViewedAt: unreadChan.last_viewed_at,
         deltaMsgs: delta,
+        deltaMsgsRoot: deltaRoot,
     };
 
     return data;
 }
 
-export function setUnreadPost(userId: string, postId: string) {
+export function setUnreadPost(userId: string, postId: string, location: string) {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
         let state = getState();
         const post = Selectors.getPost(state, postId);
         let unreadChan;
 
+        const collapsedThreadsEnabled = isCollapsedThreadsEnabled(state);
+        const isThreadUnread = collapsedThreadsEnabled && location === THREAD;
```

Comment:
can you have this variable be more self explanatory, right now it can be confused with CRT, perhaps something like `setUnreadFromThreadScreen` or something like that